### PR TITLE
Add burn support for only caching packages

### PR DESCRIPTION
**Pull request by heaths**
I realize we'll discuss this change in triage, but wanted to submit the PR anyway after migrating the patch over to the wix4 repo and in case we take it as-is.

### CODE REVIEW

**Comment by barnson** [comment]
Is there a different PR with your port? This one's old.

**Comment by heaths** [comment]
You're right. I thought I had done a newer port without the command line switch. I'll push that soon.

**Comment by heaths** [comment]
This is now ready. Commit 2f485c5 made the same changes as I did to wix3 to move /cache support from engine to BA.

File: History.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+* ErnestT: WIXBUG:4432 - Add burn support for only caching packages
+
```
Comment (rseanhall) [inline History.md]: Shouldn't be modifying History.md anymore.

File: src/burn/engine/plan.cpp [hunk 2]
```diff
@@ -625,7 +625,11 @@ extern "C" HRESULT PlanRegistration(
     UINT cDependencies = 0;
 
     pPlan->fRegister = TRUE; // register the bundle since we're modifying machine state.
-    pPlan->fKeepRegistrationDefault = (pRegistration->fInstalled || BOOTSTRAPPER_RESUME_TYPE_REBOOT == resumeType); // keep the registration if the bundle was already installed or we are planning after a restart.
+
+    // Keep the registration if the bundle was already installed or we are planning after a restart.
+    // Also keep the registration if we are caching because the ARP (via uninstall) allows uncaching.
```
Comment (rseanhall) [inline src/burn/engine/plan.cpp]: "Also keep the registration..." should be removed, right?  This comment isn't in wix3.

File: src/burn/engine/plan.cpp [hunk 3]
```diff
@@ -1063,16 +1067,21 @@ extern "C" HRESULT PlanCachePackage(
...
     if (fBARequestedCache || NeedsCache(pPlan, pPackage))
     {
+        // The behavior for cache only mode is to do nothing on rollback, e.g. for subsequent install on demand scenarios.
+        pPackage->rollback = BOOTSTRAPPER_ACTION_STATE_NONE;
+
         hr = AddCachePackage(pPlan, pPackage, phSyncpointEvent);
         ExitOnFailure(hr, "Failed to plan cache package.");
 
+        // Update plan state to account for the package being cached
+        pPlan->qwEstimatedSize += pPackage->qwSize;
```
Comment (rseanhall) [inline src/burn/engine/plan.cpp]: Per #108, we don't touch pPlan->qwEstimatedSize here.  This needs to be removed.

**Comment by rseanhall** [comment]
My personal preference is not to have merge commits in a pull request, it makes a lot of noise in the history.

File: src/burn/engine/plan.cpp (hunk 3 above)
Comment (heaths) [inline src/burn/engine/plan.cpp]: I'm guessing this needs to be changed in wix3 as well? Seems qwEstimatedSize is updated elsewhere there as well.

Comment (rseanhall) [inline src/burn/engine/plan.cpp]: No, #108 is wix4 only.

**Comment by robmen** [comment]
This has been merged but Github doesn't seem to be recognizing so. Closing manually.

**Pull request closed without merging**